rust/signer/src/lib.rs

```diff
@@ -89,11 +93,73 @@ fn qrcode_bytes(data: &[u8]) -> Option<String> {
         pixels: &pixels,
         width: qr.size() as usize,
         scale: 16,
-    }.render(&mut result).ok()?;
+    }
+    .render(&mut result)
+    .ok()?;
 
     Some(base64png(&result))
 }
 
+fn construct_suri(seed: &str, path: &str, password: &str) -> String {
+    format!("{}{}///{}", seed, path, password)
+}
+
+#[derive(Debug, PartialEq)]
+struct SURI {
+    phrase: String,
+    path: String,
+    password: String,
+}
+
+fn parse_suri(suri: &str) -> Result<SURI, String> {
+    fn parse_derivation_path(input: &str) -> Result<(String, String), String> {
+        let re = Regex::new(r"^((?:\/\/?[^/]+)*)(?:\/\/\/(.*))?$").expect("invalid regex");
```

Comment:
what is `parse_suri` function used for? is it only used for substrate seed?

what is the difference with the one in `suri.js`, 

If not, this regex does not catch the regex of brain wallet seed with extra white space in this [PR](https://github.com/paritytech/parity-signer/pull/467), e.g '  good//sdsadsa/sdsa'

Reply:
It is identical to the one in `suri.js` (prior to your PR).  The point is to replicate logic so that the seed/suri can be used correctly in rust without returning to js.

> If not, this regex does not catch the regex of brain wallet seed with extra white space in this PR, e.g ' good//sdsadsa/sdsa'

I am updating it, along with other grumbles.

Reply:
I see, I forget what it is in the previous `suri.js`, but this one seems like the regex from `parseDerivationPath`, which is not the `parseSuri` one.

Reply:
Yes, the regex above is from `parseDerivationPath`, which is here nested as `parse_derivation_path` inside `parse_suri`.  The other one below should be identical to the regex from `parseSURI` in suri.js (before your recent changes).  The overall logic should be the same.

Reply:
ah, yeah, I missed the nested function.